src/sql/parts/notebook/notebook.contribution.ts

```diff
@@ -44,6 +49,11 @@ export class NewNotebookAction extends Action {
 		let title = `Untitled-${counter++}`;
 		let untitledUri = URI.from({ scheme: Schemas.untitled, path: title });
 		let model = new NotebookInputModel(untitledUri, undefined, false, undefined);
+		if(!model.providerId)
+		{
+			let notebookRegistry = Registry.as<INotebookProviderRegistry>(Extensions.NotebookProviderContribution);
+			model.providerId = notebookRegistry.getProviderForFileType(DEFAULT_NOTEBOOK_FILETYPE);
```

Comment:
Thanks, this looks great. Note that my separate PR removes this class and adds a MainThreadNotebookDocumentsAndEditors that has similar logic. I think I need to take this code and put there - please check this in and I'll handle merge conflicts.